dogen/generator.py

```diff
@@ -289,11 +291,27 @@ def handle_sources(self):
                 target = basename
 
             filename = ("%s/%s" % (self.output, target))
+
             passed = False
+            algorithm = None
+
+            md5sum = source.get('md5sum')
+
+            # Backwards compatibility for md5sum
+            if md5sum and not source.get('md5'):
+                self.log.warn("The 'md5sum' key is deprecated, please use 'md5' for %s. Or better switch to sha1 or sha256." % url)
+                source['md5'] = md5sum
+
+            for supported_algorithm in SUPPORTED_HASH_ALGORITHMS:
+                if not source.get(supported_algorithm):
+                    continue
+
+                algorithm = supported_algorithm
```

Comment:
Yes. Do you think we need to change it? We select the most secure one. Isn't sufficient? Or maybe a warning? WDYT?